ckanext/qdes/logic/action/get.py

```diff
@@ -0,0 +1,341 @@
+                rows.append({
+                    'Dataset name': package.get('title', package.get('name', '')),
+                    'Link to dataset (URI)': url_for('dataset.read', id=package.get('id'), _external=True),
+                    'Resource name': resource.get('name', ''),
+                    'Link to resource': url_for('resource.read',
+                                                resource_id=resource.get('id'),
+                                                id=package.get('id'),
+                                                package_type=package.get('type'),
+                                                _external=True
+                                                ),
+                    'Dataset creator': package.get('contact_creator', ''),
+                    'Point of contact - name':
+                        _qdes_extract_point_of_contact(package.get('contact_point', None), 'Name'),
+                    'Point of contact - email':
+                        _qdes_extract_point_of_contact(package.get('contact_point', None), 'Email'),
+                    'List of recommended fields without values': ', '.join(missing_values),
+                    'Organisation name': package.get('organization').get('title', ''),
+                })
+
+    return rows
+
+
+def _get_uri_validated_fields(scheme, field_group):
+    uri_validated_fields = []
+    for field in scheme.get(field_group, []):
+        validator = field.get('validators', '').split()
+        if 'qdes_uri_validator' in validator:
+            uri_validated_fields.append(field)
+
+    return uri_validated_fields
+
+
+def qdes_datasets_with_invalid_urls(context, config):
+    u"""
+    List of all datasets with broken links to resources.
+    """
+    # Get list of invalid uris.
+    uris = Session.query(InvalidUri).all()
+    invalid_uris = [uri.as_dict() for uri in uris]
+
+    # Build package list.
+    entities = {}
+    for invalid_uri in invalid_uris:
+        if invalid_uri.get('entity_id') in entities:
+            entities[invalid_uri.get('entity_id')].get('fields').append(invalid_uri.get('field'))
+        else:
+            entities[invalid_uri.get('entity_id')] = {}
+            entities[invalid_uri.get('entity_id')].update({
+                'type': invalid_uri.get('entity_type'),
+                'fields': [invalid_uri.get('field')],
+            })
+
+    # Build rows.
+    rows = []
+    packages = {}
+    resources = {}
+    for entity_id in entities:
+        # Load entity and cache them.
+        entity_dict = {}
+        parent_entity_dict = {}
+        invalid_uri = entities.get(entity_id)
+        if invalid_uri.get('type') == 'dataset':
+            if entity_id in packages:
+                entity_dict = packages.get(entity_id)
+            else:
+                entity_dict = get_action('package_show')({}, {'id': entity_id})
+
+                if config and config.get('org_id') and entity_dict.get('organization').get('id') != config.get(
```

Comment:
Hi @awset - `config.get('org_id')` is used multiple times in this function... if we default set `config={}` in the function declaration, then set `org_id = config.get('org_id', None)` at the start of this function, we can just refer to `org_id` var within the function and reduce the `if config and config.get('org_id')` checks to `if org_id`

Reply:
I will implement it for other functions